src/GitVersion.App.Tests/TagCheckoutInBuildAgentTests.cs

```diff
@@ -0,0 +1,60 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using GitTools.Testing;
+using GitVersion.BuildAgents;
+using GitVersion.Core.Tests.Helpers;
+using LibGit2Sharp;
+using NUnit.Framework;
+using Shouldly;
+
+namespace GitVersion.App.Tests
+{
+    [TestFixture]
+    public class TagCheckoutInBuildAgentTests
+    {
+        [Test]
+        public async Task VerifyTagCheckoutOnGitHubActions()
+        {
+            var env = new Dictionary<string, string>
+            {
+                { GitHubActions.EnvironmentVariableName, "true" },
+                { "GITHUB_REF", "ref/tags/0.2.0" },
+            };
+
+            using var fixture = new EmptyRepositoryFixture();
+            var remoteRepositoryPath = PathHelper.GetTempPath();
+            RepositoryFixtureBase.Init(remoteRepositoryPath);
+            using (var remoteRepository = new Repository(remoteRepositoryPath))
+            {
+                remoteRepository.Config.Set("user.name", "Test");
+                remoteRepository.Config.Set("user.email", "test@email.com");
+                fixture.Repository.Network.Remotes.Add("origin", remoteRepositoryPath);
+                Console.WriteLine("Created git repository at {0}", remoteRepositoryPath);
+                remoteRepository.MakeATaggedCommit("0.1.0");
+                Commands.Checkout(remoteRepository, remoteRepository.CreateBranch("develop"));
+                remoteRepository.MakeACommit();
+                Commands.Checkout(remoteRepository, remoteRepository.CreateBranch("release/0.2.0"));
+                remoteRepository.MakeACommit();
+                Commands.Checkout(remoteRepository, "main");
```

Comment:
```suggestion
                Commands.Checkout(remoteRepository, TestBase.MainBranch);
```

Reply:
Doh! Missed that one, thanks for catching that!  Fixed up.